cpu-miner.c

```diff
@@ -573,8 +628,40 @@ static bool gbt_work_decode(const json_t *val, struct work *work)
 	bin2hex(work->txs, txc_vi, n);
 	bin2hex(work->txs + 2*n, cbtx, cbtx_size);
 
+	// VeriBlock:
+	// read keystones
+        keystones = json_object_get(val, "keystone_hashes");
+        if (!keystones || !json_is_array(keystones)) {
+          applog(LOG_ERR, "JSON invalid keystone_hashes");
+          goto out;
+        }
+        keystone_count = json_array_size(keystones);
+        if(keystone_count != keystones_total) {
+          applog(LOG_ERR, "Expected to get %d keystones, but got %d", keystones_total, keystone_count);
+          goto out;
+        }
+        for (i = 0; i < keystone_count; i++) {
+          const json_t *ks = json_array_get(keystones, i);
+          const char *ks_hex = json_string_value(ks);
+          if (!ks_hex || !hex2bin(unauthcontext + k, ks_hex, 32)) {
+            applog(LOG_ERR, "JSON invalid keystone at index %d", i);
+            goto out;
+          }
+          k += 32; // sizeof(keystone hash)
+        }
+        
+    const int keystoneSize = 32;
+    for (k=0; k<2; k++)
+    {
+        unsigned char *ks_begin = unauthcontext + 4 + k*keystoneSize;
+        tool_invert_buffer(ks_begin, keystoneSize);
+    }
+
 	/* generate merkle root */
-	merkle_tree = malloc(32 * ((1 + tx_count + 1) & ~1));
+    unsigned long mtreeSize = 32 * ((1 + tx_count + 1) & ~1);
+	merkle_tree = malloc(mtreeSize);
```

Comment:
do not use tabs, use only spaces

Reply:
all project uses tabs
I delete my spaces and inserted tabls as same as project